docusaurus/docs/iOS/client/controllers/channels.md

```diff
@@ -54,28 +54,43 @@ let compoundFilter: Filter<ChannelListFilterScope> = .and([
 
 The `query.sort` is an array of sorting options. Sorting options are applied based on their order in the array so the first option has the highest impact while the others are used mainly as a tiebreakers. By default, the channel list is sorted by `updated_at`.
 
+:::note
+If your query is filtering based on custom data, please [make sure you pass a filter block when creating a controller](#%EF%B8%8F-when-the-filter-cannot-be-auto-resolved).
+:::
+
 ### 2. Create a controller
 
-The controller can be created using `ChatClient`:
+The controller can be created using an instance of `ChatClient`:
 ```swift
-let controller = ChatClient.shared.channelListController(query: query, filter: { channel in
-    // the filtering logic goes here.
-    return true
+let client = ChatClient.shared
+let controller = client.channelListController(query: query)
+```
+
+The filter passed in `query.filter` is going to be used to filter out the channels that do not belong to the list when receiving updates.
+
+#### :warning: When the filter cannot be auto-resolved:
+
+In certain edge cases, it can be that we cannot resolve at runtime the filter you passed in the query. In that case, you will either see an assertion or a log in the console pointing you to this documentation.
+
+We have a fallback that can be used when this happens. The method that is used to create a `ChatChannelListController` also accepts a `filter` block
+```swift
+let client = ChatClient.shared
+let controller = client.channelListController(query: query, filter: { channel in
+    // The filtering logic goes here.
+    return true/false
```

Comment:
We should provide a concrete example in this case, otherwise, customers are really not sure what to expect and what to write. We need to provide a real example, with a real query, and a real closure.